jgrapht-core/src/main/java/org/jgrapht/alg/scoring/BetweenesCentrality.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * (C) Copyright 2017-2017, by Assaf Mizrachi and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.scoring;
+
+import java.util.*;
+
+import org.jgrapht.*;
+import org.jgrapht.alg.interfaces.*;
+import org.jgrapht.util.*;
+
+/**
+ * Betweenes centrality.
+ * 
+ * <p>
+ * Computes the betweenes centrality of each vertex of a graph. The betweenness centrality of a node
+ * $v$ is given by the expression: $g(v)= \sum_{s \neq v \neq
+ * t}\frac{\sigma_{st}(v)}{\sigma_{st}}$ where $\sigma_{st}$ is the total number of shortest paths
+ * from node $s$ to node $t$ and $\sigma_{st}(v)$ is the number of those paths that pass through
+ * $v$. For more details see
+ * <a href="https://en.wikipedia.org/wiki/Betweenness_centrality">wikipedia</a>.
+ * 
+ * The algorithm is based on 
+ * <ul>
+ * <li>randes, Ulrik (2001). "A faster algorithm for betweenness centrality". 
+ * Journal of Mathematical Sociology. 25 (2): 163–177.</li>
+ * </ul>
+ *
+ * The running time is $O(nm) and $O(nm +n^2 \log n)$ for unweighted and weighted graph respectively, 
+ * where $n$ is the number of vertices and $m$ the number of edges of the graph. The space
+ * complexity is $O(n + m)$.
+ *
+ * 
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ * 
+ * @author Assaf Mizrachi
+ * @since December 2017
+ */
+public class BetweenesCentrality<V, E>

```

Comment:
The class name (and all references to it) should be spelled out in full:  "betweenness" not "betweenes".